### PR TITLE
EHN reorganize the way to read data

### PR DESCRIPTION
**Pull request by glemaitre**

### CODE REVIEW

File: build_tools/appveyor/test.ps1 [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# installed library.
+- "cd C:\\"
+
+$installed_specio_folder = $(python -c "import os; os.chdir('c:/'); import specio;\
+print(os.path.dirname(specio.__file__))")
+echo "specio found in: $installed_specio_folder"
+
+# --pyargs argument is used to make sure we run the tests on the
+# installed package rather than on the local folder
+py.test --pyargs specio $installed_specio_folder
```
Comment (jorisvandenbossche) [inline build_tools/appveyor/test.ps1]: I think this is a simpler approach: https://github.com/uwescience/shablona/blob/master/appveyor.yml

**Comment by glemaitre** [comment]
@jorisvandenbossche could you have a look. I think that we could merge now.

File: appveyor.yml [hunk 2]
```diff
@@ -26,35 +28,11 @@ install:
...
-  # Install other specio dependencies
-  - "python setup.py bdist_wheel"
-  - ps: "ls dist"
-
-  # Install the generated wheel package to test it
-  - "pip install --pre --no-index --find-links dist/ specio"
-
-# Not a .NET project, we build in the install step instead
-build: false
+  - conda install pip numpy six nose wheel pytest pytest-cov -y -q
```
Comment (jorisvandenbossche) [inline appveyor.yml]: nose still needed?

File: appveyor.yml [hunk 3]
```diff
@@ -26,35 +28,11 @@ install:
...
 
 test_script:
-  - powershell ./build_tools/appveyor/test.ps
-
-artifacts:
-  # Archive the generated packages in the ci.appveyor.com build report.
-  - path: dist\*
+  - mkdir for_test
+  - cd for_test
+  - py.test --pyargs specio --cov-report term-missing --cov=specio
```
Comment (jorisvandenbossche) [inline appveyor.yml]: py.test -> pytest

File: appveyor.yml (hunk 2 above)
Comment (jorisvandenbossche) [reply appveyor.yml]: wheel also not?

File: specio/plugins/tests/test_example.py [hunk 4]
```diff
@@ -22,5 +22,6 @@ def test_dummy_format():
     assert reader.get_length() == 1
     assert reader.get_meta_data() == {}
     spec = reader.get_data(0)
+    print(spec)
```
Comment (jorisvandenbossche) [inline specio/plugins/tests/test_example.py]: leftover?

File: specio/core/format.py [hunk 5]
```diff
@@ -395,23 +394,19 @@ def _get_data(self, index=None):
             It should return the spectra and meta data: (ndarray, dict).
```
Comment (jorisvandenbossche) [inline specio/core/format.py]: Update the docstring here that it now returns Spectrum or list of them

Comment (jorisvandenbossche) [reply specio/core/format.py]: Just thinking, the index can be a bit two things: if you have a single spectrum object with multiple spectra, give the nth spectrum, but if you a list of spectrum objects, give the nth object (which can still have multiple spectra).

**Pull request merged**